### PR TITLE
#6 add weekend filter

### PR DESCRIPTION
**Pull request by rrudol**
## Description:
Added weekend filter

## Github Issue: 
https://github.com/akai-org/hackathons/issues/6

## Screenshots: 
image

### CODE REVIEW

File: src/hackathons-collector/hackathons-collector.js [hunk 1]
```diff
@@ -24,31 +25,48 @@ import akai from "./assets/akai-logo.svg";
...
           <Logo src={logo} />
           <H1>Hackathons Collector</H1>
         </Header>
+        <Filter
+          value={this.state.filterWeekends}
+          setValue={newValue => {
+            this.setState({ filterWeekends: newValue });
+          }}
+        >
+          Only weekend events
```
Comment (roertbb) [inline src/hackathons-collector/hackathons-collector.js]: Filter component does not take children prop 😉We could pass that and make it quite reusable

File: src/hackathons-collector/hackathons-collector.js [hunk 2]
```diff
@@ -24,31 +25,48 @@ import akai from "./assets/akai-logo.svg";
...
+    this.state = {
+      filterWeekends: false,
+      allEvents: events
+    };
+  }
+  render() {
+    const previousEvents = this.state.allEvents
       .filter(event => event.date <= new Date())
+      .filter(event => {
+        return !this.state.filterWeekends || event.date.getDay() > 5;
```
Comment (roertbb) [inline src/hackathons-collector/hackathons-collector.js]: We could extract that condition, that's used in `previousEvents` and `events`, but I'm fine with how it looks now 😉

Comment (rrudol) [reply src/hackathons-collector/hackathons-collector.js]: I've tried to figure it out using functional programming paradigm, but naming these structures was very tricky so, for now, let's stay with that solution 😉

**Comment by roertbb** [comment]
I've added one little improvement mentioned somewhere above and make it finally shipped 😅

**Review by roertbb — APPROVED** [review]

**Pull request merged**